### PR TITLE
Upgrade to akka-http and akka-stream 2 and added more acceptance tests

### PR DESCRIPTION
**Pull request by danpersa**
Added new acceptance tests
Upgraded to akka http and akka streams 2

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala [hunk 1]
```diff
@@ -81,7 +82,7 @@ class Routes @Inject() (implicit val materializer: ActorMaterializer,
                     LOG.info("post regex /routes/")
                     handleWith(saveRoute)
                   }
-                } ~ (hasOneOfTheScopes(authenticatedUser)(scopes.WRITE_FULL_PATH)) {
+                } ~ (hasOneOfTheScopes(authenticatedUser)(scopes.WRITE_STRICT, scopes.WRITE_REGEX) & isStrictRoute(route.route)) {
```
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala]: fixed bug discovered during writing the acceptance tests :)

File: src/it/scala/org/zalando/spearheads/innkeeper/AcceptanceSpecsHelper.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  val uri = "http://localhost:8080/routes"
+  implicit val defaultPatience =
+    PatienceConfig(timeout = Span(2, Seconds), interval = Span(500, Millis))
+  implicit val system = ActorSystem("main-actor-system")
+  implicit val materializer = ActorMaterializer()
+
+  def entityString(response: HttpResponse): String = {
+    response.entity.dataBytes
+      .map(bs => bs.utf8String)
+      .runFold("")((a, b) => a + b)
```
Comment (mo-gr) [inline src/it/scala/org/zalando/spearheads/innkeeper/AcceptanceSpecsHelper.scala]: Code golf: could probably be something like `.runFold("")(_+_)`

File: src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala (hunk 1 above)
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala]: That's what they are for :+1:

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala [hunk 3]
```diff
@@ -34,15 +35,18 @@ class OAuthService @Inject() (val config: Config,
   override def authorize(token: String): Try[AuthorizedUser] = {
     import org.zalando.spearheads.innkeeper.oauth.OAuthJsonProtocol.authorizedUserFormat
 
-    val response = Http().singleRequest(HttpRequest(uri = Uri(url(token))))
+    val responseFuture = Http().singleRequest(HttpRequest(uri = Uri(url(token))))
 
-    val futureJson = Source(response.map {
-      _.entity.dataBytes.map(bs => bs.utf8String)
-    }).flatten(FlattenStrategy.concat).runFold("")(_ + _)
+    val futureFutureJson: Future[Future[String]] = (responseFuture.map { res =>
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: `Future[Future[String]]`? I think you want to `flatMap` here

File: src/it/scala/org/zalando/spearheads/innkeeper/AcceptanceSpecsHelper.scala (hunk 2 above)
Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/AcceptanceSpecsHelper.scala]: :+1:

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala (hunk 3 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: I'm doing it in the next for...

File: build.sbt [hunk 4]
```diff
@@ -55,4 +54,6 @@ ScalariformKeys.preferences := ScalariformKeys.preferences.value
   .setPreference(DoubleIndentClassDeclaration, true)
   .setPreference(AlignParameters, true)
 
+parallelExecution in IntegrationTest := false
```
Comment (danpersa) [inline build.sbt]: without this, there is a total crash of the acceptance tests...

**Comment by mo-gr** [comment]
lgtm. feel free to merge after travis is green

**Pull request merged**